core/validation/failures.py

```diff
@@ -390,6 +427,9 @@ def serialise_user_centered_failures(
         # ignore tab and section for pre-transformation failures
         return {"PreTransformationErrors": [message for _, _, message in uc_failures]}
 
+    uc_failures = list(set(uc_failures))
+    uc_failures.sort()
```

Comment:
Could you add a comment to just mention that this is to remove duplicate messages